src/goto-symex/renaming_level.h

```diff
@@ -51,13 +59,44 @@ struct symex_renaming_levelt
   }
 };
 
+/// Wrapper for expressions or types which have been renamed up to a given
+/// \p level
+template <typename underlyingt, levelt level>
+class renamedt
```

Comment:
`underlyingt` appears to be `ssa_exprt` everywhere -- simply hardwire that?

Reply:
For now it is, but my plan is to next use it with `exprt` (draft PR here: https://github.com/diffblue/cbmc/pull/3986 though it needs to be rebased on this one), we could also think of using it with typet